src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -223,9 +223,16 @@ private static bool IsSelfClosing(Block beginTagBlock)
 
             if (name == null)
             {
-                errorSink.OnError(span.Start,
-                                  RazorResources.TagHelperBlockRewriter_TagHelperAttributesMustBeWelformed,
-                                  span.Content.Length);
+                // We couldn't find a name, if the original span content was whitespace it ultimately means the tag
+                // that owns this "attribute" is malformed and is expecting a user to type a new attribute.
+                // ex: <myTH class="btn"| |
+                if (!string.IsNullOrWhiteSpace(span.Content))
+                {
+                    errorSink.OnError(
+                        span.Start,
+                        RazorResources.TagHelperBlockRewriter_TagHelperAttributesMustBeWelformed,
+                        span.Content.Length);
+                }
```

Comment:
how is this change related to #311 ?  if it's a quick fix for an unrelated issue, just get it in -- separately.